e2e-tests/src/test/adder.rs

```diff
@@ -19,26 +18,22 @@ pub async fn adder() -> Result<()> {
 
     let (conn, _authority, account) = basic_test_context(config).await?;
 
-    let contract = Arc::new(AdderInstance::new(
+    let contract = AdderInstance::new(
         &config.test_case_params.adder,
         &config.test_case_params.adder_metadata,
-    )?);
-
-    let listen_conn = conn.clone();
-    let listen_contract = contract.clone();
-    let (tx, mut rx) = unbounded();
-    let listen = || async move {
-        listen_contract_events(&listen_conn, &[listen_contract.as_ref().into()], tx).await?;
-        <Result<(), anyhow::Error>>::Ok(())
-    };
-    let join = tokio::spawn(listen());
```

Comment:
Request: Write a new test instead of changing this code. This is not code testing the adder contract - the adder contract is used for testing the client. As far as I can tell you're removing the tests for event listening and replacing them with the tests for this new mechanism.

Reply:
I agree - I've made two testcases, but decided not to extract common setup checks, since it requires abstracting some action, that takes inner action and returns an event, which makes things quite illegible

however, if you have some suggestion, I'm eager to fix this

Reply:
Looks good